src/app/beer_garden/requests.py

```diff
@@ -596,12 +637,19 @@ def process_request(
 
     try:
         logger.info(f"Publishing request {request.id}")
-        queue.put(
-            request,
-            confirm=True,
-            mandatory=True,
-            delivery_mode=pika.spec.PERSISTENT_DELIVERY_MODE,
-        )
+        # This is how you publish requests when they are connected to the local garden.
+        if request.garden_name == get_local_garden_name():
+            queue.put(
+                request,
+                confirm=True,
+                mandatory=True,
+                delivery_mode=pika.spec.PERSISTENT_DELIVERY_MODE,
+            )
+        # If it is not controlled locally, we need to forward it
+        else:
+            raise RoutingRequestException(
+                f"Unable to route request {request.id}, {request.garden_name} is not hosted on {get_local_garden_name()}"
+            )
```

Comment:
Looks like I forgot to push this back up to the routing class... Here is a question, I put this in here to protect us when updating code later. If we botch up our code and we let things get forwarded down to this level, then there could be a chance that it still gets routed back down to this code and now we are in an infinite loop. Should we just throw the error and I remove the comments?

Reply:
I don't think it's a good idea to be planning for failure in that sense. If we mess up the routing logic to the point where things are routing incorrectly then we need better error checking in the router. Once it gets to this point we have to assume it's been routed correctly.

If we decide that routing a request incorrectly is something that 100% can't happen then that sounds like a case for lots of unit tests :smile: 

Even in the case where this is routed incorrectly I don't think we'd get into an infinite loop. I think we'd attempt to publish to a queue that doesn't exist (because from a "wrong" garden's perspective, it doesn't), which would just fail and raise.